### PR TITLE
Add Accessibility Considerations section

### LINKED ISSUE #1557: Request for an Accessibility Considerations section to API for Accessing Public key credentials CR
An accessibility review was requested of the APA as part of our role in performing a horizontal review of W3C documents for accessibility concerns.

We reviewed Web Authentication: An API for accessing Public Key Credentials Level 2 W3C Candidate Recommendation Snapshot:  https://www.w3.org/TR/webauthn-2/

The following comment (https://lists.w3.org/Archives/Public/public-apa/2020Dec/0021.html) was reviewed and approved via a CfC by the APA working group (https://lists.w3.org/Archives/Public/public-apa/2021Feb/0029.html):

We have concerns that could be best summarized in a new section "Accessibility Considerations" which could follow "Security Considerations" or "Privacy Considerations" in document order. References to timing considerations should be updated to reference this new subheading. See editor's draft https:/w3c.github.io/webauthn/. Additionally, based on the accessibility topics below, notes could be added to the appropriate sections (e.g., registration).

Proposed topics for "Accessibility Considerations":
1. Public key credentials should avoid using a single biometric factor. We would also like to call your attention to the W3C Note, Inaccessibility of CAPTCHA, Alternatives to Visual Turing Tests on the Web (https://www.w3.org/TR/turingtest/). 
2. Registration should provide affordances for users to complete authorization gestures correctly. This could involve naming the authenticator, choosing a picture to associate with the device, or entering freeform text instructions.
3. Ceremonies that rely on timing must follow WCAG Guideline 2.2 Enough Time (https://www.w3.org/WAI/WCAG21/Understanding/enough-time).

Thank you for your consideration.

### PR DESCRIPTION
**Pull request by equalsJeffH**
Per feedback in issue #1557

fixes #1557

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1558.html" title="Last updated on Feb 17, 2021, 9:13 PM UTC (aba7dd6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1558/e6791c0...aba7dd6.html" title="Last updated on Feb 17, 2021, 9:13 PM UTC (aba7dd6)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -7052,6 +7052,14 @@ the [=[RP]=] could mitigate the privacy leak using the same approach of returnin
 as discussed in [[#sctn-username-enumeration]].
 
 
+# Accessibility Considerations # {#sctn-accessiblility-considerations}
+
+[=Client platforms=], at [=registration=] time, should provide affordances for users to complete [=authorization gestures=] correctly. This could involve naming the authenticator, choosing a picture to associate with the device, or entering freeform text instructions.
+
+[=Ceremonies=] relying on timing, e.g., during registration (see {{PublicKeyCredentialCreationOptions/timeout}}), ought to follow [WCAG Guideline 2.2 Enough Time](https://www.w3.org/WAI/WCAG21/Understanding/enough-time).
```
Comment (lgarron) [inline index.bs:7059]: The API allows specifying a timeout. If the RP-provided timeout is in conflict with this line (e.g. something very short like 5 seconds), should the user agent be empowered to extend the ceremony timeout?

Comment (equalsJeffH) [reply index.bs:7059]: I think so (?) given [WCAG Guideline 2.2 Enough Time] ?  Tho would be good for others with more expertise in this particular aspect of W3C specs to weigh-in ?

**Comment by nadalin** [comment]
@agl @emlun @akshayku Please review as soon as you can

**Review by nadalin — APPROVED** [review]

File: index.bs (hunk 1 above)
Comment (agl) [reply index.bs]: Note that, in practice, browsers already bound the RP-provided timeout to sane values.

Comment (equalsJeffH) [reply index.bs]: ( doh, shudda looked at the code... )

So I interpret that as: adding that 2nd sentence is OK

Comment (equalsJeffH) [reply index.bs]: done in 00a5760

**Review by agl — APPROVED** [review]

File: index.bs (hunk 1 above)
Comment (becka11y) [reply index.bs]: Yes, thanks for adding that additional sentence/clarification.

**Comment by equalsJeffH** [comment]
@becka11y -- does this PR satisfactorily address your issue #1557 ?

**Comment by becka11y** [comment]
We didn't get a chance to review this in the APA meeting today. It will be on the agenda for next week (2/17) if that is acceptable. If not, please let me know and we can discuss on our mailing list.

File: index.bs [hunk 2]
```diff
@@ -7052,6 +7052,13 @@ the [=[RP]=] could mitigate the privacy leak using the same approach of returnin
 as discussed in [[#sctn-username-enumeration]].
 
 
+# Accessibility Considerations # {#sctn-accessiblility-considerations}
+
+[=Client platforms=], at [=registration=] time, should provide affordances for users to complete [=authorization gestures=] correctly. This could involve naming the authenticator, choosing a picture to associate with the device, or entering freeform text instructions.
+
+[=Ceremonies=] relying on timing, e.g., during a [=registration ceremony=] (see {{PublicKeyCredentialCreationOptions/timeout}}) or during an [=authentication ceremony=] (see {{PublicKeyCredentialRequestOptions/timeout}}), ought to follow [[!WCAG21]]'s [Guideline 2.2 Enough Time](https://www.w3.org/TR/WCAG21/#enough-time). If a [=client platform=] determines that a [=[RP]=]-supplied timeout does not appropriately adhere to the latter [[!WCAG21]] guidelines, then the [=client platform=] MAY adjust the timeout accordingly.
```
Comment (lgarron) [inline index.bs:7059]: Teeny drive-by nit suggestion: you could simplify the grammar by taking out some words:
```suggestion
[=Ceremonies=] relying on timing, e.g., a [=registration ceremony=] (see {{PublicKeyCredentialCreationOptions/timeout}}) or an [=authentication ceremony=] (see {{PublicKeyCredentialRequestOptions/timeout}}), ought to follow [[!WCAG21]]'s [Guideline 2.2 Enough Time](https://www.w3.org/TR/WCAG21/#enough-time). If a [=client platform=] determines that a [=[RP]=]-supplied timeout does not appropriately adhere to the latter [[!WCAG21]] guidelines, then the [=client platform=] MAY adjust the timeout accordingly.
```

**Comment by equalsJeffH** [comment]
I added the notion of having more than one UV method for UV-capable authenticators.

**Review by selfissued — APPROVED** [review]
LGTM

**Comment by equalsJeffH** [comment]
on 2021-02-17 call:
@akshay & @agl note that in 2nd parag that "client platform" ought to be "relying party"

File: index.bs [hunk 3]
```diff
@@ -7052,6 +7052,15 @@ the [=[RP]=] could mitigate the privacy leak using the same approach of returnin
 as discussed in [[#sctn-username-enumeration]].
 
 
+# Accessibility Considerations # {#sctn-accessiblility-considerations}
+
+[=User verification=]-capable [=authenticators=], whether [=roaming authenticators|roaming=] or [=platform authenticators|platform=], should offer users more than one user verification method.  For example, both fingerprint sensing and PIN entry. This allows for fallback to other user verification means if the selected one is not working for some reason. Note that in the case of [=roaming authenticators=], the authenticator and platform might work together to provide a user verification method such as PIN entry [[FIDO-CTAP]].
```
Comment (emlun) [inline index.bs:7057]: Should the "should"s throughout the section be "SHOULD"s? I see there's a normative "MAY" in the last sentence.

File: index.bs [hunk 4]
```diff
@@ -7052,6 +7052,15 @@ the [=[RP]=] could mitigate the privacy leak using the same approach of returnin
...
+
+[=Client platforms=], at [=registration=] time, should provide affordances for users to complete [=authorization gestures=] correctly. This could involve naming the authenticator, choosing a picture to associate with the device, or entering freeform text instructions (e.g., as a reminder-to-self).
```
Comment (emlun) [inline index.bs:7059]: >[=Client platforms=], at [=registration=] time, should provide affordances for users to complete [=authorization gestures=] correctly.

Does this mean authorization gestures in future ceremonies? I first read this as applying only to registration ceremonies, not authentication ceremonies. Would something like this be in line with your intent?

```suggestion
[=Client platforms=], at [=registration=] time, should provide affordances for users to complete future [=authorization gestures=] correctly. This could involve naming the authenticator, choosing a picture to associate with the device, or entering freeform text instructions (e.g., as a reminder-to-self).
```

File: index.bs (hunk 3 above)
Comment (equalsJeffH) [reply index.bs:7057]: I think not in the 1st parag since that is with regards to authenticators themselves and this is not an authenticator-oriented spec.

I think yes in 2nd parag, since that is wrt Relying Parties.

**Comment by equalsJeffH** [comment]
Accessible Platform Architectures (APA) met today and their minutes do not mention their horizontal review of WebAuthn, even tho in https://github.com/w3c/webauthn/pull/1558#issuecomment-777050812 (above) the thought was that it would be on their agenda today. 

Given all that, and @akshayku s[aying on today's call that he's good with this being merged upon fixing the item noted in](https://www.w3.org/2021/02/17-webauthn-minutes.html) https://github.com/w3c/webauthn/pull/1558#issuecomment-780823352 (above), we're merging this.
